tgui/packages/tgui/interfaces/ChemHeater.js

```diff
@@ -0,0 +1,123 @@
+
+export const ChemHeater = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { containerData, isActive, targetTemperature } = data;
+
+  return (
+    <Window
+      title="Reagent Heater/Cooler"
+      width={320}
+      height={385}>
+      <Window.Content>
+        <ChemDisplay container={containerData} targetTemperature={targetTemperature} active={isActive} />
+        <Section title="Temperature Control">
+          <Stack align="center">
+            <Stack.Item>
+              <Knob
+                animated
+                size={2}
+                value={targetTemperature}
+                minValue={0}
+                maxValue={1000}
+                format={value => value + " K"}
+                onDrag={(e, value) => act('adjustTemp', { temperature: value })}
+              />
+            </Stack.Item>
+            <Stack.Item grow basis={0} overflow="hidden">
+              <Box
+                className="ChemHeater__TemperatureNumber"
+                nowrap
+                p={1}
+                fontSize={1.5}
+                color={getTemperatureColor(targetTemperature)}
+                backgroundColor="black">
+                <Box fontSize={1}>Target</Box>
+                <Icon name={getTemperatureIcon(targetTemperature)} pr={0.5} />
+                <AnimatedNumber value={targetTemperature} /> K
+              </Box>
+            </Stack.Item>
+            <Stack.Item basis={9.6} align="center">
+              <Button
+                icon="power-off"
+                disabled={!containerData?.totalVolume}
+                color={isActive ? "red" : "green"}
+                fluid
+                height="100%"
+                fontSize={1.25}
+                textAlign="center"
+                onClick={() => act(isActive ? 'stop' : 'start')}>
+                {isActive ? "Deactivate" : "Activate"}
+              </Button>
+            </Stack.Item>
+          </Stack>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const ChemDisplay = (props, context) => {
+  const { act } = useBackend(context);
+  const { active = false, targetTemperature = freezeTemperature } = props;
+  const container = props.container ?? NoContainer;
+  const working = active && !container.fake;
+  const { temperature, totalVolume } = container;
+
+  return (
+    <SectionWithTitleOverflow
+      title={
+        <Box as="span"
+          style={{
+            "text-transform": "capitalize",
+          }}>
+          {container.name}
+        </Box>
+      }
+      buttons={
+        <Button
+          icon="eject"
+          disabled={!props.container}
+          onClick={() => act('eject')}>
+          Eject
+        </Button>
+      }>
+      <ReagentGraph container={container} />
+      <ReagentList container={container} />
+      <Box className={"ChemHeater__TemperatureBox " + (working && `ChemHeater__TemperatureBox__${getTemperatureChangeName(temperature, targetTemperature)}`)}>
```

Comment:
Should use `classes` util function to avoid adding a `false` class or whatever the default behavior is here (JavaScript) if `working` is `false`.